src/app/shared/components/page-not-found/page-not-found.component.ts

```diff
@@ -0,0 +1,17 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'page-not-found',
+  templateUrl: './page-not-found.component.html',
+  styleUrls: ['./page-not-found.component.scss'],
+})
+export class PageNotFoundComponent implements OnInit {
+  constructor(private route: Router) {}
+
+  public backToHomePage(): void {
+    this.route.navigate(['products']);
```

Comment:
Keep in mind, that 404 page is global. So in could redirrect not only on the products page